modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/jsonld/UriJsonLDSupport.scala

```diff
@@ -0,0 +1,44 @@
+package ch.epfl.bluebrain.nexus.kg.indexing.jsonld
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.kg.indexing.IndexingVocab.JsonLDKeys._
+import io.circe.Json
+
+/**
+  * Typeclass definition that adds JSON-LD support into [[Uri]]
+  *
+  */
+trait UriJsonLDSupport {
```

Comment:
you can just create an implicit class to turn a Uri into a Json representation with an `@id` key.

Reply:
Done